main/src/library/Graph.flix

```diff
@@ -101,4 +101,68 @@ namespace Graph {
         Reachable(n1, n2) :- Reachable(n1, m), Edge(m, n2).
     }
 
+    ///
+    /// Optionally returns the shortest distance from `src` to `dst` in the weighted directed graph `g`.
+    ///
+    pub def distance(src: {src :: t}, dst: {dst :: t}, g: m[(t, d, t)]): Option[d] with Foldable[m], Boxable[t], Boxable[d], Add[d] =
+        let edges = inject g into Edge;
+        let x = query edges, dists(src.src) select x from Distance(n, x) where n == dst.dst;
+        List.minimum(x)
+
+    ///
+    /// Returns the shortest distance from `src` to every other vertex in the weighted directed graph `g`.
+    ///
+    pub def distances(src: t, g: m[(t, d, t)]): Map[t, Option[d]] with Foldable[m], Boxable[t], Boxable[d], Add[d] =
+        let edges = inject g into Edge;
+        let allDists = #{
+            // Collects all the distances `d` from `src` to `n` into a Set.
+            AllDists(n; Set#{d}) :- Distance(n, d).
```

Comment:
This will be easier if `Distance` is a lattice predicate, although it's a nice solution for the current setup.